api/src/main/java/org/openmrs/module/locationbasedaccess/utils/LocationUtils.java

```diff
@@ -80,6 +85,31 @@ public static String getUserAccessibleLocationUuid(User authenticatedUser) {
         return accessibleLocationUuid;
     }
 
+    /**
+     * Used to get the accessible locations for the user. It will first get from the user property. if the user
+     * property is not available, then check for the locations having user privilege. If both are not available then return null.
+     * @param authenticatedUser Authenticated user
+     * @return accessible Location uuid
+     */
+    public static List <String> getUserAccessibleLocationUuidsWithUserPrivileges(User authenticatedUser){
+        if (authenticatedUser == null) {
+            return null;
+        }
+        List <String> accessibleLocationUuids = new ArrayList<String>();
+        String accessibleLocationUuid = authenticatedUser.getUserProperty(LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
+        if (StringUtils.isBlank(accessibleLocationUuid)) {
+            List<Location> openMrsLocations = Context.getLocationService().getAllLocations();
```

Comment:
@suthagar23 the function `getUserAccessibleLocationUuidsWithUserPrivileges` is called from the `LocationServiceInterceptorAdvice` when the function is matched. here we are checking if privilege matches with any of the Location by `getAllLocations`  so it is being called again so this is intercepted by aop interceptor it is being recursive thereby giving a StackOverflow error.

Reply:
What could be the solution then?

Reply:
Is their a way to call this from Daemon thread.?